Family analysis (#17)
populationgenomics/automated-interpretation-pipeline: design_docs/MOI_Checks.md

4. Where a variant passes all conditions within a filter class, a 'result' object is created
5. The result of `.run()` is a list of valid modes of inheritance (ReportedVariant), each including details of the
variant(s), and samples

### Family Checks

> Instead of a tree-traverse of families, the method used in family checks here is to represent each family as a flat pool

Reviewer: ```suggestion
Instead of a tree-traverse of families, the method used in family checks is to represent each family as a flat pool
```

of participants. The inheritance rules are applied unilaterally to every member of a family group, rather than in a
directional manner, e.g. when considering each candidate variant:

- for a monoallelic variant, inherited with complete penetrance, we enforce a unilateral rule - every person with the
variant must also be affected. If this rule is violated for any individual member, the variant does not pass the
inheritance test.
- for a biallelic variant pair, all participants with the variant pair must also be affected. If the variant considered
is Homozygous, we permit unaffected members to be heterozygous. Similarly, for compound-het inheritance, unaffected
participants may have either variant but not both.

> Code has also been written to enable Complete and Incomplete penetrance models to be used. For the Complete penetrance

Reviewer: ```suggestion
Complete and Incomplete penetrance models to be used are used in this module. For the Complete penetrance
```
